cookbooks/bach_ambari/attributes/default.rb

```diff
@@ -0,0 +1,45 @@
+node.force_default['ambari']['install_java'] = false
+node.force_default['ambari']['java_home'] = '/usr/lib/jvm/java-8-oracle-amd64/'
+
+
+# FILES
+namenodes = get_namenodes.select{ |nn| nn[:bcpc][:node_number] }.map{ |nn| "namenode#{nn[:bcpc][:node_number]}" }.join(',')
```

Comment:
Please do not call `get_namenodes`; the Chef node search code is a costly thing to execute. We have a function `set_hosts` which updates a number of node attributes to use instead. Please use the [node attributes](https://github.com/bloomberg/chef-bach/blob/e4352ff2604bf14b19cee62fffb045fc6e321e54/cookbooks/bcpc-hadoop/libraries/utils.rb#L186-L200) defined there.